Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -853,6 +867,29 @@ AXTextMarker AXTextMarker::findMarker(AXDirection direction, AXTextUnit textUnit
                 else if (boundary == AXTextUnitBoundary::End && start <= end && end != -1 && end >= objectBorder)
                     resultMarker = AXTextMarker(currentObject->treeID(), currentObject->objectID(), end - objectBorder);
             }
+        };
+
+        auto updateSentenceResultMarker = [&] () {
+            if (boundary == AXTextUnitBoundary::Start) {
+                int start = previousSentenceStartFromPosition(flattenedRuns, offset);
+                if (direction == AXDirection::Previous && start < objectBorder && start != -1)
+                    resultMarker = AXTextMarker(currentObject->treeID(), currentObject->objectID(), start);
+                else if (direction == AXDirection::Next && start != -1 && start >= objectBorder)
+                    resultMarker = AXTextMarker(currentObject->treeID(), currentObject->objectID(), start - objectBorder);
+            } else {
+                int end = nextSentenceEndFromPosition(flattenedRuns, offset);
+                if (direction == AXDirection::Previous && end <= objectBorder && end != -1)
+                    resultMarker = AXTextMarker(currentObject->treeID(), currentObject->objectID(), end);
+                else if (direction == AXDirection::Next && end != -1 && end >= objectBorder)
+                    resultMarker = AXTextMarker(currentObject->treeID(), currentObject->objectID(), end - objectBorder);
```

Comment:
A good follow-up change to this PR would be making a new `AXTextMarker` constructor that takes an `AXCoreObject` and an offset so we aren't repeating `AXTextMarker(currentObject->treeID(), currentObject->objectID(), offset)` all over the place (instead all these callsites could look like `AXTextMarker(currentObject, offset)`)

Reply:
I was noticing that as well... I can definitely follow up with that!